src/pypromice/process/L2toL3.py

```diff
@@ -18,81 +24,134 @@ def toL3(L2, T_0=273.15, z_0=0.001, R_d=287.05, eps=0.622, es_0=6.1071,
-           
-        rho_atm_l = 100 * p_h_l / R_d / (T_h_l + T_0)                          # Calculate atmospheric density                                  
-        nu_l = calcVisc(T_h_l, T_0, rho_atm_l)                                 # Calculate kinematic viscosity  
-        q_h_l = calcHumid(T_0, T_100, T_h_l, es_0, es_100, eps,                # Calculate sp.humidity
-                           p_h_l, RH_cor_h_l)
-        if not ds.attrs['bedrock']:       
-            SHF_h_l, LHF_h_l= calcHeatFlux(T_0, T_h_l, Tsurf_h, rho_atm_l, WS_h_l, # Calculate latent and sensible heat fluxes 
-                                            z_WS_l, z_T_l, nu_l, q_h_l, p_h_l)        
-            SHF_h_l, LHF_h_l = cleanHeatFlux(SHF_h_l, LHF_h_l, T_h_l, Tsurf_h, p_h_l, # Clean heat flux values
-                                              WS_h_l, RH_cor_h_l, ds['z_boom_l'])
-            ds['dshf_l'] = (('time'), SHF_h_l.data)
-            ds['dlhf_l'] = (('time'), LHF_h_l.data)
-        q_h_l = 1000 * q_h_l                                                   # Convert sp.humid from kg/kg to g/kg
-        q_h_l = cleanSpHumid(q_h_l, T_h_l, Tsurf_h, p_h_l, RH_cor_h_l)         # Clean sp.humid values
-        ds['qh_l'] = (('time'), q_h_l.data)    
+            if ('wspd_l' in ds.keys()) and \
+                ('t_surf' in ds.keys()) and \
+                    ('z_boom_l' in ds.keys()):
+                z_WS_l = ds['z_boom_l'].copy() + 0.4                                   # Get height of W                  
+                z_T_l = ds['z_boom_l'].copy() - 0.1                                    # Get height of thermometer 
+                WS_h_l = ds['wspd_l'].copy()                                 
+                if not ds.attrs['bedrock']:       
+                    SHF_h_l, LHF_h_l= calcHeatFlux(T_0, T_h_l, Tsurf_h, WS_h_l, # Calculate latent and sensible heat fluxes 
+                                                    z_WS_l, z_T_l, q_h_l, p_h_l)        
+        
+                    ds['dshf_l'] = (('time'), SHF_h_l.data)
+                    ds['dlhf_l'] = (('time'), LHF_h_l.data)
+            else:
+                logger.info('wspd_l, t_surf or z_boom_l missing, cannot calulate tubrulent heat fluxes')
+    
+            q_h_l = 1000 * q_h_l                                                       # Convert sp.humid from kg/kg to g/kg
+            ds['qh_l'] = (('time'), q_h_l.data)
+        else:
+            logger.info('t_l, p_l or rh_l_cor missing, cannot calulate tubrulent heat fluxes')
 
+    # Smoothing and inter/extrapolation of GPS coordinates
+    for var in ['gps_lat', 'gps_lon', 'gps_alt']:
+        ds[var.replace('gps_','')] = gpsCoordinatePostprocessing(ds, var)
+
+    # adding average coordinate as attribute        
+    for v in ['lat','lon','alt']:
+        ds.attrs[v+'_avg'] = ds[v].mean(dim='time').item()
     return ds
 
+def gpsCoordinatePostprocessing(ds, var, config_folder='../aws-l0/metadata/station_configurations/'):
```

Comment:
I see. Can we make it at once for all scripts in a separate PR? Because having hybrid standard (when gradually changing from one to the other) is even more confusing.